draft-rescorla-tls-esni.md

```diff
@@ -274,6 +274,12 @@ period. Servers SHOULD set the Resource Record TTL small enough so that the
 record gets discarded by the cache before the ESNI keys reach the end of
 their validity period.
 
+Client MAY cache the ESNIKeys for a particular domain based on the TTL of the
+Resource Record, but MUST NOT cache it based on the not_after value, to allow
+servers to rotate the keys often and improve forward secrecy (due to clock skew
+on the client, it may not be possible for the server to use a very short key
+lifetime).
```

Comment:
I am sympathetic to the proposed change, but might prefer using "SHOULD NOT" rather than a "MUST NOT" here.

Consider a client that uses `getaddrinfo`. The client cannot access the TTL value. OTOH, it makes sense for such a client to cache the query result for a short amount of time to reduce the load on the resolver running on the client machine. My understanding is that some clients are implemented that way.

Reply:
This PR only prevents the client from using "not_after" for caching, it doesn't really forbid using another random value (not sure if we want to be more specific on this).

Anyway, I'm fine changing this to SHOULD NOT.